### PR TITLE
sentry-native: add crashpad support

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **sentry-native/all**

Add crashpad support to sentry-native.
Depends on #5511 when using with `-o crashpad:backend=crashpad`.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by espkk** [comment]
This looks nice, but IIRC sentry-native uses [own crashpad fork](https://github.com/getsentry/crashpad)
It basically allows sentry_options_add_attachment to be used. Without it crashpad backend vs raw crashpad does not make much sense...
Am I missing something?

**Comment by madebr** [comment]
@espkk 
You're right. attachment support is a feature of getsentry's fork.
These are the changes between chromium's crahpad and getsentry's: https://github.com/getsentry/crashpad/compare/master..getsentry

I just looked at the release page of sentry-native, and it looks like they provide a `sentry-native.zip` file with everything included.
https://github.com/getsentry/sentry-native/releases

**Comment by espkk** [comment]
> I just looked at the release page of sentry-native, and it looks like they provide a `sentry-native.zip` file with everything included.
> https://github.com/getsentry/sentry-native/releases

That's nice! I was sure it is shipped without submodules.
I believe
`self._cmake.definitions["SENTRY_CRASHPAD_SYSTEM"] = True` should be removed then

**Comment by madebr** [comment]
@espkk 
I'm working on a `sentry-crashpad` recipe at https://github.com/conan-io/conan-center-index/pull/5536

**Comment by espkk** [comment]
> 
> 
> @espkk
> I'm working on a `sentry-crashpad` recipe at #5536

I see... But could you please explain why we need it if it's anyway downloaded in the source bundle? 
Considering that the fork already makes it cmake-compatiable I do not understand that additional dependency

**Comment by madebr** [comment]
Increased modularity.
You can use crashpad from multiple sources: the official sources (this pr) or a fork (sentry's one).

**Comment by espkk** [comment]
> Increased modularity.
> You can use crashpad from multiple sources: the official sources (this pr) or a fork (sentry's one).

Ok, that makes sense.
Then may I ask you one more question since I'm not familar enough with conan yet.
How in practice can I switch between 'the official sources (this pr) or a fork (sentry's one)'?
I mean, I would expect some package:option that switches SENTRY_CRASHPAD_SYSTEM (and adds/removes self.requires)
Could you please show an example of configuration?

**Comment by madebr** [comment]
How to switch transparently (or from the command line or using profiles), that I don't know because I will hardcode `sentry-crashpad` in the requirement list in the `sentry-native` recipe.

By using [`provides`](https://docs.conan.io/en/latest/reference/conanfile/attributes.html#provides), conan will complain if you try to mix multiple crashpad packages.

File: recipes/sentry-native/all/conanfile.py [hunk 1]
```diff
@@ -3,6 +3,8 @@
 from conans import ConanFile, CMake, tools
 from conans.errors import ConanInvalidConfiguration
 
+required_conan_version = ">=1.28.0"
```
Comment (SpaceIm) [inline recipes/sentry-native/all/conanfile.py:6]:
```suggestion
required_conan_version = ">=1.33.0"
```

**Comment by jgsogo** [comment]
There are some recipes that are already modifying the requirements based on some option, you can find examples related to the different JPEG implementations https://github.com/conan-io/conan-center-index/blob/41d4fbb8c218f9d938ccb8b9d888bff52002e62e/recipes/libyuv/all/conanfile.py#L40

**Review by jgsogo — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Comment by Dev-vlad** [comment]
Hi @madebr! I saw that you are working on a recipe for the sentry-crashpad. Are there any plans to integrate it with the current one?
I'm not sure how conan ci works, but I'm getting a build error on windows: 
`sentry_backend_crashpad.cpp(254,31): error C2039: 'SetFirstChanceExceptionHandler': is not a member of 'cra
shpad::CrashpadClient'`
I've checked the generated projects and the version of `crashpad` used in the current recipe is not compatible with `sentry-native` on windows.

**Comment by madebr** [comment]
@Dev-vlad 
Can you open an issue + a small reproducer cmake project?
I tested the test package on Linux and it worked.
sentry's fork of crashpad (`sentry-crashpad`) has also become available on cci,
so I'll add an option to the `sentry-native` recipe to switch between `crashpad` and `sentry-crashpad`.

**Comment by madebr** [comment]
@Dev-vlad 
I could reproduce your issue by building sentry-native with MSVC. "It works" on Linux 😄 .
Soon, I'll open a pr.

**Comment by Dev-vlad** [comment]
@madebr yep, this issue is only for Windows. I checked sentry-crashpad and they have a different API under `#ifdef (OS_WIN)` that is not available in the original crashpad project.

**Comment by madebr** [comment]
@Dev-vlad 
See https://github.com/conan-io/conan-center-index/pull/5717

**Pull request merged**